hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -605,12 +619,76 @@ private void dispatchSyntheticTxn(
             dispatcher.dispatchHandle(childContext);
             childRecordBuilder.status(ResponseCodeEnum.SUCCESS);
             childStack.commitFullStack();
-        } catch (HandleException e) {
+        } catch (final HandleException e) {
             childRecordBuilder.status(e.getStatus());
             recordListBuilder.revertChildrenOf(childRecordBuilder);
         }
     }
 
+    private record ValidationResult(
+            @NonNull PreHandleResult.Status status, @NonNull ResponseCodeEnum responseCodeEnum) {}
+
+    private ValidationResult validate(
+            @NonNull final KeyVerifier verifier,
+            final HederaFunctionality function,
+            final TransactionBody txBody,
+            final AccountID payer,
+            final Key payerKey,
+            final TransactionCategory txCategory) {
+
+        final PreHandleContextImpl preHandleContext;
+        try {
+            preHandleContext =
+                    new PreHandleContextImpl(readableStoreFactory(), txBody, payer, configuration(), dispatcher);
+            dispatcher.dispatchPreHandle(preHandleContext);
+        } catch (final PreCheckException e) {
+            return new ValidationResult(PRE_HANDLE_FAILURE, e.responseCode());
+        }
+
+        // Check if the payer has the required permissions
+        if (!authorizer.isAuthorized(payer, function)) {
+            if (function == HederaFunctionality.SYSTEM_DELETE) {
+                return new ValidationResult(PRE_HANDLE_FAILURE, ResponseCodeEnum.NOT_SUPPORTED);
+            }
+            return new ValidationResult(PRE_HANDLE_FAILURE, ResponseCodeEnum.UNAUTHORIZED);
+        }
+
+        // Check if the transaction is privileged and if the payer has the required privileges
+        final var privileges = authorizer.hasPrivilegedAuthorization(payer, function, txBody);
+        if (privileges == SystemPrivilege.UNAUTHORIZED) {
+            return new ValidationResult(PRE_HANDLE_FAILURE, ResponseCodeEnum.AUTHORIZATION_FAILED);
+        }
+        if (privileges == SystemPrivilege.IMPERMISSIBLE) {
+            return new ValidationResult(PRE_HANDLE_FAILURE, ResponseCodeEnum.ENTITY_NOT_ALLOWED_TO_DELETE);
+        }
+
+        // Skip payer verification when dispatching a child transaction
+        if (!txCategory.equals(CHILD)) {
```

Comment:
@tinker-michaelj  is this going to be enough to skip the payer key verification?

Reply:
@stoyanov-st This should work for the `ContractService`, and is good for now I think. 🤔

Reply:
Does this also work for `ScheduleService`?

Reply:
This might help with some of the issues in schedule transactions.  Most of the time Scheduled transactions should follow a very similar pattern to a synthetic transaction; although there are some important differences (e.g. top level record instead of child, transaction ID matched to create rather than the triggering transaction, etc...).
I'll try implementing the same in the paths scheduled transactions exercise and see how it goes.

Reply:
This code path should be executed for scheduled transactions, too. (The method name is misleading.) We should not implement separate dispatch mechanisms and validations for each possible use case we will have.

Reply:
It's not about verify, it's more about whether this will fix some issues in the schedule execution specs.
I'll follow up separately from this PR.

Reply:
Agreed.. @stoyanov-st  Please change them in follow up PR as Joseph suggested.